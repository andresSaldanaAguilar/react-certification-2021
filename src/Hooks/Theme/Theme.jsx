import React, { createContext, useContext, useState } from 'react';

const ThemeStateContext = createContext();
const ThemeDispatchContext = createContext();

function ThemeProvider({ children }) {
  const [theme, setTheme] = useState('light');
  return (
    <ThemeStateContext.Provider value={theme}>
      <ThemeDispatchContext.Provider value={setTheme}>
        {children}
      </ThemeDispatchContext.Provider>
    </ThemeStateContext.Provider>
  );
}
function useThemeState() {
  const context = useContext(ThemeStateContext);
  if (context === undefined) {
    throw new Error('useThemeState must be used within a ThemeProvider');
  }
  return context;
}
function useThemeDispatch() {
  const context = useContext(ThemeDispatchContext);
  if (context === undefined) {
    throw new Error('useThemeDispatch must be used within a ThemeProvider');
  }
  return context;
}
export { ThemeProvider, useThemeState, useThemeDispatch };
