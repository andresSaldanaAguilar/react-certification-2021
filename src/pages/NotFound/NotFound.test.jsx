import React from 'react';
import { render } from '@testing-library/react';
import NotFoundPage from './NotFound.page';
import Theme from '../../components/App/Theme.styled';

describe('Not Found Landing View', () => {
  it('Should render 404 not found message', () => {
    const { getByText } = render(
      <Theme>
        <NotFoundPage />
      </Theme>
    );
    expect(getByText('404 Nothing Found')).toBeInTheDocument();
  });
});
