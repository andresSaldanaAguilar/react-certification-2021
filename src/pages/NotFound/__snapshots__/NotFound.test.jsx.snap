// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Landing View Should render 404 not found message 1`] = `
<DocumentFragment>
  <section
    class="not-found"
  >
    <div
      class="MuiContainer-root sc-bdfBwQ esBNfU MuiContainer-maxWidthSm"
    >
      <h1
        class="MuiTypography-root sc-gsTCUz iNyAAg MuiTypography-h1"
      >
        404 Nothing Found
      </h1>
    </div>
  </section>
</DocumentFragment>
`;
